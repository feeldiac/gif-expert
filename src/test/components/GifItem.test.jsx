const { render, screen } = require("@testing-library/react")
import GifItem from "../../components/GifItem"

const title = 'Lady Gaga'
const url = 'https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExdW1ncWdoaHFkMXdvMWJjZXRrOHdkNjlqNDZpanduYnY4MXNjeGtlOCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/TjGFDxbbZRYjv9vpCL/giphy.gif'

describe('Testing GifItem', () => {
    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('should display the image with the url', () => {
        render(<GifItem title={title} url={url} />)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('should display the title', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})