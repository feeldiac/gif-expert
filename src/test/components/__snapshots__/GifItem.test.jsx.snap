// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Lady Gaga"
      src="https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExdW1ncWdoaHFkMXdvMWJjZXRrOHdkNjlqNDZpanduYnY4MXNjeGtlOCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/TjGFDxbbZRYjv9vpCL/giphy.gif"
    />
    <p>
      Lady Gaga
    </p>
  </div>
</div>
`;
